Bluetooth: Mesh: add mesh frame statistic module (#59962)
zephyrproject-rtos/zephyr: doc/connectivity/bluetooth/api/mesh/statistic.rst

.. _bluetooth_mesh_stat:

> Statistic

Reviewer: Frame statistics?

#########

> The statistic API allows to monitor number of received frames over
> different interfaces. Also, the statistic module allows to monitor
> the number of planned and succeeded transmission and relaying attempts.

Reviewer: ```suggestion
The statistic API allows monitoring the number of received frames over
different interfaces, and the number of planned and succeeded
transmission and relaying attempts.
```

> It helps API user to estimate the efficiency of the advertiser configuration
> parameters as well as the scanning ability of the device. Number of
> the monitored parameters can be easily extended by customer values.

Reviewer: ```suggestion
The API helps the user to estimate the efficiency of the advertiser configuration
parameters and the scanning ability of the device. The number of the monitored
parameters can be easily extended by customer values.
```
